ow_power_system/src/power_system_node.cpp

```diff
@@ -106,6 +106,59 @@ vector<map<MessageId, Datum<double>>> PowerSystemNode::loadPowerProfile(const st
   return result;
 }
 
+bool PowerSystemNode::loadFaultPowerProfiles()
+{
+  string path;
+  path = ros::package::getPath("ow_power_system") + "/data/low_state_of_charge_power_failure.csv";
+  m_low_state_of_charge_power_failure_sequence = loadPowerProfile(path);
+
+  path = ros::package::getPath("ow_power_system") + "/data/instantaneous_capacity_loss_power_failure.csv";
+  m_instantaneous_capacity_loss_power_failure_sequence = loadPowerProfile(path);
+
+  path = ros::package::getPath("ow_power_system") + "/data/thermal_power_failure.csv";
+  m_thermal_power_failure_sequence = loadPowerProfile(path);
+
+  return true;
+}
+
+bool PowerSystemNode::initPrognoser()
+{
+  // Create a configuration from a file
+  auto prognoser_config_path = ros::package::getPath("ow_power_system") + "/config/prognoser.cfg";
+  ConfigMap prognoser_config(prognoser_config_path);
+
+  // Contruct a new prognoser using the prognoser factory. The prognoser
+  // will automatically construct an appropriate model, observer and predictor
+  // based on the values specified in the config.
+  m_prognoser = PrognoserFactory::instance().Create("ModelBasedPrognoser", prognoser_config);
+
+  if (m_prognoser == nullptr)
+    return false;
```

Comment:
Issue a ROS error here, so the user knows what went wrong?

Reply:
There is already an error message for that on. Check line 358: `ROS_ERROR("Power system node failed to initialize");`. Of course it is not detailed but it should let the user know there was a problem during initialization. Let me know if this is not sufficient and a more detailed message is needed.

Reply:
It's fine as it is. then.  I'm guessing it's a rare problem.

Reply:
It didn't happen to me .. I just added this as an extra check